01_fib/jimbotech/ex-fibonacci.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// fibonacci returns a function that returns
```

Comment:
I can't comment on closure and it is definitely an interesting way of doing it. I personally feel if you have to write this much comment to explain what the function does, then it needs to be simplified. From a learning perspective, it might be easier to learn how it is done as using go structure and syntax.

Reply:
"Closure" is a key concept in go as per the tour of go: https://tour.golang.org/moretypes/25 I added all the comments so in six months time I can still remember why I did it. Hope it is ok if I keep it just like this.